float32 b_l0
float32 l0_l1
